### ISSUE TITLE
pymysql returning the error "pymssql._pymssql.OperationalError) (3971, b'The server failed to resume the transaction. Desc:3700000003.DB-Lib error message 20018, severity 16:\nGeneral SQL Server error: Check messages from the SQL Server\"

### ISSUE THREAD

**Issue by manmayaray** [pymssql/pymssql#901]
**Describe the bug**
Due to the DB error main API is throwing 500 Internal service error when connecting to DB. Recently upgraded to 2.2.7. Earlier we are using 2.1.5 with this services were going down and unable to receover completely.
**To Reproduce**
Hitting multiple times or even once making the insert to go into DB error.

**Expected behavior**
System should work as expected with DB error.
**Current Behavior**
Currently at engine level this is set.
db = SQLAlchemy()

def init_app(app):
    db_connection_string = app.config["DB_CONNECTION_STRING"]
    app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
    app.config['SQLALCHEMY_BINDS'] = {
        'db': db_connection_string
    }
    app1.config['SQLALCHEMY_DATABASE_URI']=db_connection_string
    app.1config['SQLALCHEMY_POOL_RECYCLE'] = 299
    app1.config['SQLALCHEMY_ENGINE_OPTIONS'] = {
        "poolclass": QueuePool,
        "pool_size": 1000,
        "pool_pre_ping": True,
        "pool_recycle": 300,
        "max_overflow": 1000,
    }

**Context (Environment)**
 - What are the OS - Linux , Python and FreeTDS versions?
 - please include the output of the following command:
    `/<path to your python>/python -c "import pymssql; print(pymssql.version_info())"`

**Additional context**
  db.session.add(header)
            db.session.commit()
except Exception as e:
            LOG.error("Error while creating node")
            LOG.error(e, exc_info=True)
            raise InternalServerError("Error while processing new claim")

**Comment by manmayaray** [comment]
At moment nealy for single claim 500 rows are insrted into the DB table. Python version is 3.8.5

**Comment by manmayaray** [comment]
pymssql._pymssql.OperationalError) (20020, b'DB-Lib error message 20020, severity 9:\nBad token from the server: Datastream processing out of sync

pymssql._pymssql.OperationalError) (3971, b'The server failed to resume the transaction. Desc:3700000003.DB-Lib error message 20018, severity 16:\nGeneral SQL Server error: Check messages from the SQL Server\n

**Comment by termim** [comment]
My guess is that `"pool_size": 1000` is too high.
Are you sure you want 1000 simultaneous connections open to the single database?

**Comment by manmayaray** [comment]
Yes, my issue is that if when 2 claims are triggeting one claim is going into that exception

**Comment by manmayaray** [comment]
ok, earlier our DB used get slow down and application used to go down, there are some modification made to queries, Pool concept used and pymysql library was changed from 2.1.5 to 2.2.7. Now we are getting this error and it is very strange sometime it is causing the issue and somedays it is fine...The issue is more during peak hours..
 ,
